### PR TITLE
service: Add optional Ports for CEC Services redirect

### PR DESCRIPTION
**Pull request by jrajahalme**
Add optional ports field into CEC/CCEC Services redirect, which, if not empty, will limit the redirected frontends to the ones whose port number is on the ports list.

Note that we also limit the frontend protocol to TCP, as we do not yet support other protocols in our Envoy listeners.

```release-note
CiliumEnvoyConfig CRDs now support an optional 'ports' field in services objects, limiting the redirected service frontends to the ones whose port is listed.
```

### CODE REVIEW

File: pkg/ciliumenvoyconfig/cec_manager.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func (r *cecManager) addK8sServiceRedirects(resourceName service.L7LBResourceNam
 
 		// Tell service manager to redirect the service to the port
```
Comment (mhofstetter) [inline pkg/ciliumenvoyconfig/cec_manager.go:128]: nit: maybe enrich that comment that if configured, only a subset of frontend ports is forwarded to the L7 proxy.

**Review by mhofstetter — APPROVED** [review]
LGTM. thanks!

**Review by learnitall — APPROVED** [review]
API changes look good, thanks!

**Comment by jrajahalme** [comment]
Pushed to bump the crd patch version due to the new field in the CEC/CCEC CRD.

File: pkg/service/service.go [hunk 2]
```diff
@@ -203,6 +203,31 @@ type L7LBInfo struct {
...
+}
+
+// isPortMatch returns true if frontend has protocol TCP and its Port is in i.ports, or if i.ports
+// is empty.
+// 'ports' is typically short for no point optimizing the search.
+func (i *L7LBInfo) isPortMatch(fe *lb.L4Addr) bool {
+	// L7 LB redirect is only supported for TCP frontends
+	if fe.Protocol != lb.TCP {
+		return false
+	}
```
Comment (julianwiedmann) [inline pkg/service/service.go:218]: Can you please break this part out into a separate patch? Also, let's call this `isProtoAndPortMatch` or similar?

**Review by julianwiedmann — CHANGES_REQUESTED** [review]
Just one small structural change request.

File: pkg/service/service.go (hunk 2 above)
Comment (jrajahalme) [reply pkg/service/service.go:218]: Done.

**Review by julianwiedmann — APPROVED** [review]
lgtm, thanks!

**Review by youngnick — APPROVED** [review]

File: pkg/ciliumenvoyconfig/cec_manager.go (hunk 1 above)
Comment (jrajahalme) [reply pkg/ciliumenvoyconfig/cec_manager.go:128]: leaving this for future enhancement
